pallets/process-validation/src/restrictions.rs

```diff
@@ -8,21 +8,38 @@ use vitalam_pallet_traits::ProcessIO;
 
 #[derive(Encode, Decode, Clone, PartialEq)]
 #[cfg_attr(feature = "std", derive(Debug))]
-pub enum Restriction {
+pub enum Restriction<TokenMetadataKey, TokenMetadataValue>
+where
+    TokenMetadataKey: Parameter + Default + Ord,
+    TokenMetadataValue: Parameter + Default,
+{
     None,
     SenderOwnsAllInputs,
-    FixedNumberOfInputs { num_inputs: u32 },
-    FixedNumberOfOutputs { num_outputs: u32 },
+    FixedNumberOfInputs {
+        num_inputs: u32,
+    },
+    FixedNumberOfOutputs {
+        num_outputs: u32,
+    },
+    FixedMetadataValue {
+        input_index: u32,
+        metadata_key: TokenMetadataKey,
+        metadata_value: TokenMetadataValue,
+    },
 }
 
-impl Default for Restriction {
+impl<TokenMetadataKey, TokenMetadataValue> Default for Restriction<TokenMetadataKey, TokenMetadataValue>
+where
+    TokenMetadataKey: Parameter + Default + Ord,
+    TokenMetadataValue: Parameter + Default,
+{
     fn default() -> Self {
         Restriction::None
     }
 }
 
 pub fn validate_restriction<A, R, T, V>(
-    restriction: &Restriction,
+    restriction: &Restriction<T, V>,
```

Comment:
Where would you recommend passing those references? If I take them out it causes quite a few problems down the file which don't look super easy to rectify.

Reply:
Removing it seems to have most impact on tests

Reply:
Yes so that line would read

```rust
 restriction: Restriction<T, V>,
```

There would be multiple very simple changes to tests and 1 small change to `lib.rs`